wemulate/ext/utils/restore.py

```diff
@@ -0,0 +1,45 @@
+import wemulate.ext.utils.common as common
+import wemulate.core.database.utils as dbutils
+import wemulate.utils.tcconfig as tcutils
+
+from wemulate.core.database.models import INCOMING, OUTGOING
+
+
+def restore_device() -> None:
+    """
+    Restores all connections and tcparameters stored in the database.
+
+    Returns:
+        None
+    """
+    for connection in dbutils.get_connection_list():
+        restore_connection(connection.connection_name)
+
+
+def restore_connection(connection_name: str) -> None:
+    """
+    Restores a specific connection with its parameters stored in the database.
+
+    Args:
+        connection_name: Name of the connection which should be restored
+
+    Returns:
+        None
+    """
+    connection, current_parameter_in_db = common.get_current_applied_parameters(
```

Comment:
Just an informal comment: In general connection could be Null, which would lead to a fatal exception and the execution would abort. I think we have to consider improving exception handling in general.
Of course, this should not happen because we verify the input before but just in case...